### PR TITLE
Update Angular workspace types and add update script

### PR DESCRIPTION
**Pull request by davidlj95**
Seems Angular CLI workspace file `angular.json` types has been updated with some additions around. This adds the new types using the same `json-schema-to-typescript` approach. 

Adds a small shell script to help updating that file in the future. Let me know if JS is preffered. I'm more used to writing shell scripts, but can give try writing one using Node.js / `bun`'s compatible `child_process.exec`

### CODE REVIEW

**Comment by webpro** [comment]
Fantastic! Tried the Bash script, flawless. We could rewrite to TS, but personally I really don't mind Bash. LFG.

File: packages/knip/src/plugins/angular/update-types.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env sh
+set -eu
+angular_plugin_dir="$(cd "$(dirname "$0")" && pwd)"
```
Comment (webpro) [inline packages/knip/src/plugins/angular/update-types.sh:3]:
```suggestion
root_dir="$(git rev-parse --show-toplevel)"
angular_plugin_dir="$root_dir/packages/knip/src/plugins/angular"
```

File: packages/knip/src/plugins/angular/update-types.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+tmp_dir="$(mktemp -d)"
+cleanup() {
+  rm  -rf "$tmp_dir"
+}
+trap cleanup EXIT
+git clone --depth 1 https://github.com/angular/angular-cli "$tmp_dir"
+cd "$tmp_dir/packages/angular/cli/lib/config"
+bunx json-schema-to-typescript -i workspace-schema.json -o "$angular_plugin_dir/types.ts";
```
Comment (webpro) [inline packages/knip/src/plugins/angular/update-types.sh:11]: Knip reported unused exports. We have a few options here:

- Add as entry file (meh)
- Ignore in root `knip.json` (fine, but doesn't look super great in knip repo itself)
- Fix up unused exports using knip:

```suggestion
bunx json-schema-to-typescript -i workspace-schema.json -o "$angular_plugin_dir/types.ts";
$root_dir/packages/knip/bin/knip-bun.js --directory "$root_dir" --fix
```

**Review by webpro — COMMENTED** [review]
Noticed Knip itself complains about unused exports 😜 so added suggestions how we could deal with it, other ideas also welcome of course

File: packages/knip/src/plugins/angular/update-types.sh (hunk 1 above)
Comment (davidlj95) [reply packages/knip/src/plugins/angular/update-types.sh]: Noice! Didn't know that git magic command ✨

**Comment by webpro** [comment]
Thanks a lot David, very grateful for this 🙏

**Pull request merged**